src/pages/index.js

```diff
@@ -36,36 +45,82 @@ const editProfilePopup = new PopupWithForm(
   "#edit-profile-popup",
   handleProfileSave
 );
+const editProfileImagePopup = new PopupWithForm(
+  "#edit-profile-image-popup",
+  handleProfileImageSave
+);
 const addCardPopup = new PopupWithForm("#add-card-popup", handleAddCard);
+const confirmDeletePopup = new PopupConfirm('#confirm-delete-popup', handleDeleteCard);
 
 /*═══════╕
  │ CARDS │
  ╘═══════*/
-const cardSection = new Section(
-  { items: initialCards, renderer: handleCreateCard },
-  ".cards__list"
-);
+let cardSection;
+let targetCard;
 
 /*════════════════╕
  │ EVENT HANDLERS │
  ╘════════════════*/
 const imagePopup = new PopupWithImage("#image-popup");
+const deleteButton = document.querySelector("#confirm-delete-button");
 
 function handleAddCard({ title, url }) {
-  const card = {
-    name: title,
-    link: url,
-  };
-  cardSection.addItem(handleCreateCard(card));
+  api.addCard({name: title, link: url})
+  .then(cardObj => {
+    cardSection.addItem(cardObj);
+    addCardPopup.close();
+  })
+  .catch(err => console.error(err));
 }
 function handleCreateCard(card) {
-  return new Card(card, "#card-template", handleImageClick).getElement();
+  return new Card(card, "#card-template", handleImageClick, handleDeleteClick, handleLikeClick).getElement();
+}
+function handleDeleteCard() {
+  confirmDeletePopup.deleting(true);
+  api.deleteCard(targetCard)
+    .then(() => api.getPageInfo()
```

Comment:
Please, don't call `api.getPageInfo()` when you delete a card because it's a waste of resources on fetching, rerendering of all cards and making the server busy.


Please, delete a single card from the DOM